subprojects/common/graph/src/test/kotlin/com/avito/grapth/ShortestPathTest.kt

```diff
@@ -0,0 +1,66 @@
+package com.avito.grapth
+
+import com.avito.graph.Operation
+import com.avito.graph.ShortestPath
+import com.avito.graph.SimpleOperation
+import com.google.common.truth.Truth.assertThat
+import org.junit.jupiter.api.Test
+
+class ShortestPathTest {
+
+    @Test
+    fun `zero operations - empty path`() {
+        val longestPath = path(emptySet())
+
+        assertThat(longestPath).isEmpty()
+    }
+
+    @Test
+    fun `one operation - path with it`() {
+        val operation = SimpleOperation(id = "id")
+
+        val longestPath = path(setOf(operation))
+
+        assertThat(longestPath).containsExactly(operation)
+    }
+
+    @Test
+    fun `consecutive operations - all of them`() {
+        val a1 = SimpleOperation(id = "a1")
+        val a2 = SimpleOperation(id = "a2", predecessors = setOf("a1"))
+        val a3 = SimpleOperation(id = "a3", predecessors = setOf("a2"))
+
+        val longestPath = path(setOf(a3, a2, a1))
+
+        assertThat(longestPath).containsExactly(a1, a2, a3)
+    }
+
+    @Test
+    fun `independent chains - the shortest chain`() {
+        val a1 = SimpleOperation(id = "a1", duration = 2.0)
+        val a2 = SimpleOperation(id = "a2", duration = 3.0, predecessors = setOf("a1"))
+
+        val b1 = SimpleOperation(id = "b1", duration = 3.0)
+        val b2 = SimpleOperation(id = "b2", duration = 3.0, predecessors = setOf("b1"))
+
+        val longestPath = path(setOf(a1, a2, b1, b2))
```

Comment:
Thanks, this is just a typo. In this class, we work with the shortest path.

I mixed it up by mistake because clients use it for the longest path. The "invert" operations for it.

Reply:
Fixed in https://github.com/avito-tech/avito-android/pull/985/commits/675c094180aaa4e130576c64ef4572add497eb73